src/middlewares/mail.js

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable import/prefer-default-export */
+import { verifyEmailTemplate } from '../services/mail/template/verifyEmail';
+import { emailVerifyToken } from '../utils/index';
+import Mail from '../services/mail/Mail';
+
+
+/**
+ * @param {req} req contains the express object.
+ * @param {res} res contains the express response.
+ * @param {next} next calls the next middleware.
+ * @returns {next-middleware } returns the next middleware or throws an error.
+ * @returns {error} if any error occurs it throws an error.
+ */
+export const SendVerificationEmail = async (req, res, next) => {
+  let { email, firstName, lastName } = req.body;
```

Comment:
@zanio Good work so far. Although, you handled the errors but it not in an efficient way. Before sending the `Mail`,  you should `verify` the required fields which are `email, firstName and lastName`. But from your implementation, you just append an empty string to the parameters `''` then wait for the `send.main()` to fail before returning a message. That is not efficient please rectify.

Reply:
Done